scripts/release.py

```diff
@@ -414,62 +502,119 @@ def build_parser():
     )
-    existing.add_argument(
-        '--existing-release', default=False, action='store_true'
+    parser.add_argument(
+        '--do-not-file-docs-issue', default=False, action='store_true',
+        help='Do not create a docs issue for this release'
+    )
+    parser.add_argument(
+        '--do-not-change-files-with-release-version', default=False,
+        action='store_true',
+        help='Do not update files with the new rpc_release: "future tag"'
     )
-    parser.add_argument('--github-token')
     return parser
 
 
+def validate_args(args):
+    ''' Ensure the args are following expectations.
+    '''
+    # Store tag as a string on top of as a tag object
+    args.tag_str = str(args.tag)
+    # Future_tag must be str or None
+    if args.future_tag:
+        args.future_tag_str = str(args.future_tag)
+    # args.branch is always a string, discovered or not
+    if not args.branch:
+        branches = (b.strip() for b in
+                    git("branch", "--no-color").stdout.split("\n"))
+        for branch in branches:
+            if branch.startswith('*'):
+                args.branch = branch[2:]
+                break
+        else:
+            raise SystemExit("The current branch cannot be discovered")
+    # Github token must be in CLI or env var
+    if not args.github_token:
+        token = os.environ.get('RPC_GITHUB_TOKEN', False)
+        if token:
+            args.github_token = token
+        else:
+            raise SystemExit("Token neither found in the CLI nor in env vars")
+    # RPC version should be in CLI or env var, as a str(yes|no), default yes.
+    if args.rpc_version_check:
+        chk_state = args.rpc_version_check
+    else:
+        chk_state = os.environ.get('RPC_VERSION_CHECK', 'yes')
+    args.version_check = True if chk_state == 'yes' else False
+    return args
+
+
 def main():
+    logging.getLogger().setLevel(logging.INFO)
     parser = build_parser()
-    args = parser.parse_args()
-
-    if args.tag.rc == 1 or (not args.tag.rc and not args.tag.patch):
-        return (
-            'The script currently only supports patch releases and release '
-            'candidates after the initial one.'
-        )
+    cli_args = parser.parse_args()
+    args = validate_args(cli_args)
 
     proceed_text = (
         'WARNING - Running this script will:\n'
+        '  - Delete your cache folder (%s)\n'
         '  - add a tag to %s\n'
+        '  - update branch %s with new version number\n'
         '  - submit an issue/pull-request to %s\n'
         'If you still wish to proceed type "YES": '
-    ) % (args.repo_url, args.docs_repo_url)
+    ) % (args.cache_dir, args.repo_url, args.branch, args.docs_repo_url)
 
     proceed = raw_input(proceed_text)
 
     if proceed != 'YES':
         return
 
-    rpco_repo = Repo(url=args.repo_url, cache_dir=args.cache_dir, bare=True)
+    # Instantiate a repo.
+    if args.delete_cache_dir:
+        sh.rm(args.cache_dir, '-rf')
```

Comment:
There's always things to refactor but given this is python, lets be deliberate about using sh so this doesn't turn into a bash script in python clothing.